pallets/ocex/src/lib.rs

```diff
@@ -304,19 +304,50 @@ pub mod pallet {
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		/// On idle, use the remaining weight to do clean up, remove all ingress messages that are
 		/// older than the block in the last accepted snapshot.
-		fn on_idle(_n: BlockNumberFor<T>, remaining_weight: Weight) -> Weight {
-			// TODO: We can do it after release, as an upgrade
+		fn on_idle(_n: BlockNumberFor<T>, mut remaining_weight: Weight) -> Weight {
+			let snapshot_id = <SnapshotNonce<T>>::get();
+			while remaining_weight.ref_time() >
+				<T as Config>::WeightInfo::claim_withdraw(1).ref_time()
+			{
+				<Withdrawals<T>>::mutate(snapshot_id, |btree_map| {
+					// Get mutable reference to the withdrawals vector
+					if let Some(account) = btree_map.clone().keys().nth(1) {
+						let mut accounts_to_clean = vec![];
+						if let Some(withdrawal_vector) = btree_map.get_mut(account) {
+							if let Some(withdrawal) = withdrawal_vector.pop() {
+								if !Self::on_idle_withdrawal_processor(withdrawal.clone()) {
+									withdrawal_vector.push(withdrawal.clone());
+									Self::deposit_event(Event::WithdrawalFailed(withdrawal));
+								}
+							} else {
```

Comment:
Since `withdrawal` is `pop`ed here and `on_idle_withdrawal_processor` returns false -  `btree_map`->`withdrawal_vector` will lose this item. Is it supposed be like that?

Reply:
there's `!` prior to `Self::on_idle_withdrawal_processor()` call, which makes sure that line 319 will trigger on failure :)

Reply:
Ohh I guess missed that line... my bad =\